workflow/signal/signal.go

```diff
@@ -15,6 +17,8 @@ import (
 	"github.com/argoproj/argo-workflows/v3/workflow/common"
 )
 
+var spdyTimeout = env.LookupEnvDurationOr("SPDY_TIMEOUT", 10*time.Minute)
```

Comment:
The name of the environment variable to control this is difficult to 'discover'.

I'd suggest `POD_OUTPUT_TIMEOUT` or something along those lines. No user cares about the protocol in use.

This will also need adding to the documentation.

Reply:
Thanks for your review, i will fix it. Thanks

Reply:
`EXECUTOR_SIGNAL_TIMEOUT` was what I was thinking.

Although I'm not sure that this needs to be configurable in the first place, or as long as 10min.

It isn't entirely clear what we should do after a timeout error either though

Reply:
> It isn't entirely clear what we should do after a timeout error either though, requeueing may not fix it

Thanks for your advices,  in our environment, we observe that the pod has actually been removed, but this goroutine will still block here.

Reply:
So, in my opinion, we should set a timeout that can never happen. The original purpose of this method change in client-go was also to support timeout cancelable
![image](https://github.com/argoproj/argo-workflows/assets/76554302/0928eb50-296c-467d-a812-1bd3a711ef21)

Reply:
> So, in my opinion, we should set a timeout that can never happen. The original purpose of this method change in client-go was also to support timeout cancelable

see https://github.com/kubernetes/kubernetes/pull/103177

Reply:
> Thanks for your advices, in our environment, we observe that the pod has actually been deleted, but this goroutine will still block here.

Interesting, so a `kubectl exec` is sent to an already terminated Pod and the k8s API Server does not return?
Or the signal via `exec` works but the request does not complete for some reason?

Reply:
@fyp711 you didn't answer the above; do you know why it works that way?

Reply:
> Interesting, so a `kubectl exec` is sent to an already terminated Pod and the k8s API Server does not return? Or the signal via `exec` works but the request does not complete for some reason?
@agilgur5 I think `exec` has been executed successfully, because that pod is already deleted. I think the reason is apiserver does not return. But actually, I don't know why apiserver didn't return, I tried to analyze it, but it was very difficult. Further k8s community related submissions are this link below. You can have a look. Thank you.
https://github.com/kubernetes/kubernetes/pull/103177

Reply:
Ok, gotcha. Alan and I were trying to understand _why_ this was happening here and in https://github.com/argoproj/argo-workflows/issues/13011#issuecomment-2100244803.

I read the k8s PR, however it similarly doesn't explain _why_. It seems like possibly a k8s API Server bug or `client-go` bug and there should really be an upstream issue for this that we can point to while we use this workaround.

In the interim, I'll leave a comment here noting that it's a workaround and referencing the PR